app/services/post_alerter.rb

```diff
@@ -458,7 +458,7 @@ def push_notification(user, payload)
   def expand_group_mentions(groups, post)
     return unless post.user && groups
 
-    Group.mentionable(post.user).where(id: groups.map(&:id)).each do |group|
+    Group.mentionable(post.user, include_public: false).where(id: groups.map(&:id)).each do |group|
```

Comment:
That API feels a little weird. I think the UI should instead use something like `Group.visible` and on the backend, we rely on `Group.mentionnable` to generate the notifications.

Reply:
It feels a little bit weird and I was not happy at first either, but I have a few arguments why this makes sense: 🤔 

We already have `Group.visible_groups`, but that's not what we want here. What we want is really just mentionable groups, which sometimes can include the public ones as well. Also, it is consistent with the API for `Group.visible_groups`, which has a `include_everyone` option.